### ISSUE TITLE
Tales of Phantasia entrying battle slowly

### ISSUE THREAD

**Issue by daniel229** [hrydgard/ppsspp#8538]
Alway slow in OpenGL,slow in D3D9 since https://github.com/hrydgard/ppsspp/pull/8535

**Comment by unknownbrackets** [comment]
Well, then, now we know why it's slow in OpenGL.

I wonder if we can detect in some way that we don't need depth here...

-[Unknown]

**Comment by asbel123** [comment]
i can play it on my samsung galaxy s6 edge new version speed is better than before u need to turn off block transfer use fast foward for enter battle i use touch screen its easy for me for that game i am in dhaos castle futer and it will crash one of abilty is enemys one spell is demon summon this one will crash the game.

**Comment by asbel123** [comment]
http://aselia.wikia.com/wiki/Summon_Demon look at this website u find the enemy use demon summon its will crash the emulator.

**Comment by daniel229** [comment]
Demon summon looks fine here.
![4](https://cloud.githubusercontent.com/assets/3481559/12709426/fcacf330-c8e6-11e5-9d94-d21f9d77c319.png)

**Comment by asbel123** [comment]
thats full voice version try cross editon version

**Comment by asbel123** [comment]
u using pause magic cross editon no more magic pause at all.

**Comment by daniel229** [comment]
Still looks good.
![01](https://cloud.githubusercontent.com/assets/3481559/12814292/ee290ba2-cb78-11e5-8136-d8a9ac495aea.png)

**Comment by daniel229** [comment]
Does pause magic mean time stop,that looks fine too.Which version are you using,If not the latest one,try update it from http://buildbot.orphis.net/ppsspp/index.php , and play with default setting.

**Comment by asbel123** [comment]
are using upscale or not i am using upscale on my samsung galaxy s6 edge tales of phantasia cross edition upscale makes games crashes or not just want to know

**Comment by daniel229** [comment]
Fine with xbrz,and with muttithreaded is also good,Are you load save from gamesave not the savestate.Savestate may do thing wrong sometime.

**Comment by unknownbrackets** [comment]
Has this improved at all in Vulkan?  Still need to investigate if there's a heuristic we can apply to avoid the depth blit here.

I did have the idea of sharing depth buffers for 565 buffers, which aren't terribly common.  I wonder what this game uses...

-[Unknown]

**Comment by asbel123** [comment]
i tried it new build on my laptop battle entry on vulkan to start battle takes about 13 secs real psp if u set on cw cheat 333mhz cpu battle entry is 1 sec

**Comment by unknownbrackets** [comment]
Some portion of the problem here - the largest part on Vulkan / desktop - is caused by the limit max FPS setting.  It seems to be triggering here and causing a significant delay.

-[Unknown]

**Comment by unknownbrackets** [comment]
This is slow in OpenGL for two reasons:

 * It draws the spiral using a series of lines, each drawn in three steps (one line at a time through each of these):
   1. First, a line that sets stencil (RGB masked.)
   2. Second, a line that clears stencil when the texture is black (also RGB masked.)
   3. Finally, a stencil tested draw that actually shows on screen.
 * Because of the above, it ends up using a ton (17,000) of very small buffer objects for core contexts.  Allocating (NVIDIA) and looking up these in the std::multimap ends up being a little slow too.

Mostly it's the draw calls, though...

-[Unknown]

**Comment by unknownbrackets** [comment]
I think the multithreading has improved this situation, but it's still dropping below 30 fps on OpenGL.  Vulkan takes it pretty well.

-[Unknown]

**Comment by Gamemulatorer** [comment]
Still got slowdown during battle.
![Screenshot_20200522-064826](https://user-images.githubusercontent.com/37603562/82613632-55afca00-9bf8-11ea-9f81-6cf4cf298319.png)
### GE Dump

**Comment by unknownbrackets** [comment]
During battle is different, this game renders in some annoying ways - maybe there's a way to improve.  But the effect it uses to enter battle specifically used to be very slow, even on desktop.

-[Unknown]

**Comment by Gamemulatorer** [comment]
Can we closed this or not yet?

**Comment by Gamemulatorer** [comment]
This is also a GPU readbacks issue?

**Comment by hrydgard** [comment]
No, I don't think so. I'm not sure there's much to do here anyway.
